src/Blocks/BlockDirt.h

```diff
@@ -96,10 +94,9 @@ class cBlockDirtHandler :
 				continue;
 			}
 
-			BLOCKTYPE AboveDest;
-			NIBBLETYPE AboveMeta;
-			Chunk->GetBlockTypeMeta(BlockX, BlockY + 1, BlockZ, AboveDest, AboveMeta);
-			if (cBlockInfo::GetHandler(AboveDest)->CanDirtGrowGrass(AboveMeta))
```

Comment:
It could, if we passed it the light level, or had it use the chunk to calculate the light level.

Passing the light level wouldn't do any good, as all the method would do is wrap a check on the light level.

Passing the chunk would be inefficient as I would have to make 4 calls to find light levels instead of 3 (2 checking to see if grass can spread, 1 for checking if the current grass block can spread or if it should turn to dirt).